ui/tests/acceptance/settings/mount-secret-backend-test.js

```diff
@@ -301,26 +305,110 @@ module('Acceptance | settings/mount-secret-backend', function (hooks) {
-    assert
-      .dom('[data-test-search-select-with-modal]')
-      .doesNotExist('for type ssh, the modal field does not render.');
-    // clean up
-    await runCmd(`delete identity/oidc/key/some-key`);
+  // ARG TODO when identity_token_key is added to the internal mounts endpoint will make sure the key is set and visible on the mount config page.
+  module('WIF secret engines', function () {
+    test('it sets identity_token_key on mount config using search select list, resets after', async function (assert) {
+      // create an oidc/key
+      await runCmd(`write identity/oidc/key/some-key allowed_client_ids="*"`);
+
+      for (const engine of WIF_SECRET_ENGINES) {
+        await page.visit();
+        await page.selectType(engine);
+        await click(GENERAL.toggleGroup('Method Options'));
+        assert
+          .dom('[data-test-search-select-with-modal]')
+          .exists('Search select with modal component renders');
+        await clickTrigger('#key');
+        const dropdownOptions = findAll('[data-option-index]').map((o) => o.innerText);
+        assert.ok(dropdownOptions.includes(`some-key`), `search select options show some-key`);
+        await click(GENERAL.searchSelect.option(GENERAL.searchSelect.optionIndex('some-key')));
+        assert
+          .dom(GENERAL.searchSelect.selectedOption())
+          .hasText('some-key', 'some-key was selected and displays in the search select');
+      }
+      // Go back and choose a non-wif engine type
+      await page.back();
+      await page.selectType('ssh');
+      await click(GENERAL.toggleGroup('Method Options'));
+      assert
+        .dom('[data-test-search-select-with-modal]')
+        .doesNotExist('for type ssh, the modal field does not render.');
+      // cleanup
+      await runCmd(`delete identity/oidc/key/some-key`);
+    });
+
+    test('it allows a user with permissions to oidc/key to create an identity_token_key', async function (assert) {
+      for (const engine of WIF_SECRET_ENGINES) {
+        const path = `secrets-adminPolicy-${engine}`;
+        const secrets_admin_policy = adminPolicy();
+        const secretsAdminToken = await runCmd(
+          tokenWithPolicyCmd(`secrets-admin-${path}`, secrets_admin_policy)
+        );
+
+        await logout.visit();
+        await authPage.login(secretsAdminToken);
+        await page.visit();
+        await page.selectType(engine);
+        await page.path(path);
+        await click(GENERAL.toggleGroup('Method Options'));
+        await clickTrigger('#key');
+        // create new key
+        await fillIn(GENERAL.searchSelect.searchInput, 'new-key');
+        await click(GENERAL.searchSelect.options);
+        assert.dom('#search-select-modal').exists('modal with form opens');
+        assert.dom('[data-test-modal-title]').hasText('Create new key', 'Create key modal renders');
+
+        await click(OIDC.keySaveButton);
+        assert.dom('#search-select-modal').doesNotExist('modal disappears onCancel');
```

Comment:
I think this should be 
```suggestion
        assert.dom('#search-select-modal').doesNotExist('modal disappears onSave');
```